### ISSUE TITLE
Incompatibility with Django 3.1

### ISSUE THREAD

**Issue by JaapJoris** [Suor/django-cacheops#374]
After upgrading to Django 3.1 I experienced the following error:

```
Traceback (most recent call last):
  File "manage.py", line 24, in <module>
    execute_from_command_line(sys.argv)
  File "/usr/local/lib/python3.8/site-packages/django/core/management/__init__.py", line 401, in execute_from_command_line
    utility.execute()
  File "/usr/local/lib/python3.8/site-packages/django/core/management/__init__.py", line 377, in execute
    django.setup()
  File "/usr/local/lib/python3.8/site-packages/django/__init__.py", line 24, in setup
    apps.populate(settings.INSTALLED_APPS)
  File "/usr/local/lib/python3.8/site-packages/django/apps/registry.py", line 91, in populate
    app_config = AppConfig.create(entry)
  File "/usr/local/lib/python3.8/site-packages/django/apps/config.py", line 90, in create
    module = import_module(entry)
  File "/usr/local/lib/python3.8/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1014, in _gcd_import
  File "<frozen importlib._bootstrap>", line 991, in _find_and_load
  File "<frozen importlib._bootstrap>", line 975, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 671, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 783, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/usr/local/lib/python3.8/site-packages/cacheops/__init__.py", line 8, in <module>
    from .query import *
  File "/usr/local/lib/python3.8/site-packages/cacheops/query.py", line 17, in <module>
    from django.db.models.sql.datastructures import EmptyResultSet
ImportError: cannot import name 'EmptyResultSet' from 'django.db.models.sql.datastructures' (/usr/local/lib/python3.8/site-packages/django/db/models/sql/datastructures.py)
```

**Comment by JaapJoris** [comment]
Appearently this issue was fixed by [671e117ca](https://github.com/Suor/django-cacheops/commit/671e117ca6be345a6c93c40b2ef5229cecbcfdda) and I simply needed to update django-cacheops to the latest version :)

**Comment by sshishov** [comment]
Can we have this backported to version 4.x, please.
As we are using outdated version of redis, we cannot move to 5.x version and above.
At the same time we do not want to keep  us locked regarding Django version as well.

Best regards, Sergei.
